lizmap/www/assets/js/action.js

```diff
@@ -159,27 +164,56 @@ function addActionButton(layerId, fid, item, popupitem){
     ihtml+= '&nbsp;</button>';
     var toolbar = popupitem.next('span.popupButtonBar').find('button.popup-action[value="'+ btname + '"]');
     var popupButtonBar = popupitem.next('span.popupButtonBar');
-    if ( popupButtonBar.length != 0 ) {
-        if ( toolbar.length == 0 )
+    if (popupButtonBar.length != 0) {
+        if (toolbar.length == 0)
             popupButtonBar.append(ihtml);
         else
             toolbar.before(ihtml);
     } else {
         ihtml = '<span class="popupButtonBar">' + ihtml + '</span></br>';
         popupitem.after(ihtml);
     }
-    popupitem.find('button.btn').tooltip( {
+    popupitem.find('button.btn').tooltip({
         placement: 'bottom'
-    } );
+    });
 
     // Trigger action when clicking on button
     $('div.lizmapPopupContent button.popup-action[value="'+ btname + '"]').click(function(){
+        // Clear message
+        $('#lizmap-action-message').remove();
+
+        // Empty actionLayer: do it if button & action was active or not
+        var layer = lizMap.layers['actionLayer'];
+        layer.destroyFeatures();
+
         var val = $(this).val();
         var vals = val.split('.');
         var layerId = vals[0];
         var fid = vals[1];
         var name = vals[2];
```

Comment:
Tip: since ES6 you could replace by `const [layerId, fid, name] = vals;` 
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment